c7n/mu.py

```diff
@@ -64,13 +64,20 @@ class PythonPackageArchive(object):
     zip_compression = zipfile.ZIP_DEFLATED
 
     def __init__(self, *modules):
-        self._temp_archive_file = tempfile.NamedTemporaryFile()
+        self._temp_archive_file = tempfile.NamedTemporaryFile(delete=False)
         self._zip_file = zipfile.ZipFile(
             self._temp_archive_file, mode='w',
             compression=self.zip_compression)
         self._closed = False
         self.add_modules(None, *modules)
 
+    def __del__(self):
```

Comment:
thanks, i think we tried this before, but it looks good here as a final catch in addition to expected path cleanups, its worth noting there are a bunch of caveats on __del__ with python
https://docs.python.org/3.7/reference/datamodel.html#object.__del__